### PR TITLE
Add warning for non styled components

### PR DESCRIPTION
**Pull request by egdbear**
#2053 

Hello, i think i made a fix for showing right warning message.
Please feel free to suggest better error message :)

### CODE REVIEW

**Comment by probablyup** [comment]
Also need a test for it

**Comment by egdbear** [comment]
@probablyup Hopefully now makes more sense :)

File: src/utils/flatten.js [hunk 1]
```diff
@@ -52,6 +54,15 @@ export default function flatten(chunk: any, executionContext: ?Object, styleShee
     return `.${chunk.styledComponentId}`;
   }
 
+  /* Warn if not refring styled component */
+  if (process.env.NODE_ENV !== 'production') {
+    if (ReactIs.isElement(chunk)) {
+      return console.warn(
```
Comment (probablyup) [inline src/utils/flatten.js]: 1. I liked what you were doing before in calling the function and then checking the return to see if it was JSX. That felt more safe to me.

2. Let's not `return`, that'll cause other issues. Just the warning is fine. And let's update the text slightly:

> `${getComponentName(chunk)} is not a styled component and cannot be referred to via component selector. See https://www.styled-components.com/docs/advanced#referring-to-other-components for more details.`

`getComponentName` comes from `src/utils/getComponentName.js`

File: src/utils/test/flatten.test.js [hunk 2]
```diff
@@ -99,4 +102,22 @@ describe('flatten', () => {
     expect(flatten(['foo', func], { bool: true })).toEqual(['foo', 'static', 'bar']);
     expect(flatten(['foo', func], { bool: false })).toEqual(['foo', 'static', 'baz']);
   });
+  it('warns to refer styled-components', () => {
```
Comment (probablyup) [inline src/utils/test/flatten.test.js]: nit: can you change this test name to "warns if trying to interpolate a normal React component"

**Comment by egdbear** [comment]
@probablyup you are welcome, thanks for comments as well!

**Comment by probablyup** [comment]
Last thing, want to add a changelog message? :)

**Comment by probablyup** [comment]
Thanks again!

**Comment by mxstbr** [comment]
Thank you so much for helping us improve styled-components! Based on our [Community Guidelines](https://github.com/styled-components/styled-components/blob/master/CONTRIBUTING.md) every person that has a PR of any kind merged is offered **an invitation to the styled-components organization**—that is you right now!

Should you accept, you'll get write access to the main repository. (and a fancy `styled-components` logo on your GitHub profile!) You'll be able to label and close issues, merge pull requests, create new branches, etc. We want you to help us out with those things, so don't be scared to do them! Make sure to read our [contribution and community guidelines](https://github.com/styled-components/styled-components/blob/master/CONTRIBUTING.md), which explains all of this in more detail. Of course, if you have any questions just let me know!

**Pull request merged**